### PR TITLE
RB responsiveness 2.0

### PR DESCRIPTION
**Pull request by mvidalgarcia**
##### TODO
- [x] Investigate map issue

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
     "react-modal": "^3.9.1",
     "react-popper": "^1.3.3",
     "react-redux": "^7.1.0",
-    "react-responsive": "^7.0.0",
+    "react-responsive": "8.0.0-beta.1",
```
Comment (ThiefMaster) [inline package.json]: why not `^8.0.0-beta.1`?

File: indico/web/client/js/react/util/Responsive.jsx [hunk 2]
```diff
@@ -60,3 +60,18 @@ export default Object.assign(Responsive, {
   Tablet: factory(DIMENSIONS.tablet, DIMENSIONS.computer),
   Phone: factory(null, DIMENSIONS.tablet),
 });
+
+export function useResponsive() {
+  return {
+    isWideScreen: useMediaQuery({query: `(min-width: ${DIMENSIONS.computer}px)`}),
+    isDesktop: useMediaQuery({
+      query: `(min-width: ${DIMENSIONS.computer}px) and (max-width: ${DIMENSIONS.wideScreen -
+        1}px)`,
```
Comment (ThiefMaster) [inline indico/web/client/js/react/util/Responsive.jsx]: this wrapping here is super awful :/

File: indico/web/client/js/react/hooks.js [hunk 3]
```diff
@@ -119,3 +119,31 @@ export function useIndicoAxios({camelize, ...args}) {
...
+      prevRef.current = window.scrollY;
+      window.addEventListener('scroll', _.throttle(handleScroll, 500));
+
+      return function cleanup() {
+        window.removeEventListener('scroll', _.throttle(handleScroll, 500));
+      };
+    }
+  }, [isScrollingUp, enabled]);
+
+  return {isScrollingUp};
```
Comment (ThiefMaster) [inline indico/web/client/js/react/hooks.js]: just return a plain value instead of an object

File: indico/web/client/js/react/hooks.js [hunk 4]
```diff
@@ -119,3 +119,31 @@ export function useIndicoAxios({camelize, ...args}) {
...
+        setIsScrollingUp(prevRef.current > window.scrollY);
+        prevRef.current = window.scrollY;
+      }
+    };
+
+    if (enabled) {
+      prevRef.current = window.scrollY;
+      window.addEventListener('scroll', _.throttle(handleScroll, 500));
+
+      return function cleanup() {
```
Comment (ThiefMaster) [inline indico/web/client/js/react/hooks.js]: arrow function

File: indico/web/client/js/react/components/ResponsivePopup.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+import {Popup} from 'semantic-ui-react';
+import {useResponsive} from 'indico/react/util';
+
+const ResponsivePopup = props => {
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/ResponsivePopup.jsx]: `export default function ResponsivePopup(props) { ...`

File: indico/web/client/js/react/components/StickyWithScrollBack.jsx [hunk 6]
```diff
@@ -10,16 +10,22 @@ import PropTypes from 'prop-types';
...
+import {useResponsive} from 'indico/react/util';
+
+import {useScrollUp} from '../hooks';
 
 import './StickyWithScrollBack.module.scss';
 
-export default function StickyWithScrollBack({children, context}) {
+export default function StickyWithScrollBack({children, context, responsive}) {
   const [scrollButtonVisible, setScrollButtonVisible] = useState(false);
+  const {isScrollingUp} = useScrollUp(responsive);
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/StickyWithScrollBack.jsx]: no need for object unpacking here if you return a plain value from the hook

File: indico/web/client/js/react/util/Responsive.jsx (hunk 2 above)
Comment (ThiefMaster) [reply indico/web/client/js/react/util/Responsive.jsx]: defining

    const belowWideScreen = DIMENSIONS.wideScreen - 1;

and then using this here would avoid the wrapping

File: indico/web/client/js/react/components/ResponsivePopup.jsx (hunk 5 above)
Comment (mvidalgarcia) [reply indico/web/client/js/react/components/ResponsivePopup.jsx]: Is this because of the naming problem when importing it?

Comment (ThiefMaster) [reply indico/web/client/js/react/components/ResponsivePopup.jsx]: i think in this particular case the name would be derived correctly, but generally yes (and CRA etc. all use `function` for their top-level functions representing components)

**Comment by ThiefMaster** [comment]
The booking details tooltip on the calendar bars  should probably be disabled on all touch devices (is there a media query for it)? 

---

Can we hide more of the UI in portrait mode here? It looks a bit strange to have e.g. the legend there.

image

**Comment by ThiefMaster** [comment]
More UI to hide in portrait mode:

image

**Comment by ThiefMaster** [comment]
> Error: Invalid LatLng object: (NaN, NaN)

I guess that's because the map is hidden? I don't think this happens if the map is generally disabled, so worth investigating why this happens here.

File: indico/web/client/js/react/util/Responsive.jsx [hunk 7]
```diff
@@ -54,9 +59,58 @@ const factory = (minDimension, maxDimension) => {
...
+  };
+
+  return _OrientationSpec;
+};
+
 export default Object.assign(Responsive, {
   WideScreen: factory(DIMENSIONS.wideScreen, null),
   Desktop: factory(DIMENSIONS.computer, DIMENSIONS.wideScreen),
   Tablet: factory(DIMENSIONS.tablet, DIMENSIONS.computer),
   Phone: factory(null, DIMENSIONS.tablet),
```
Comment (ThiefMaster) [inline indico/web/client/js/react/util/Responsive.jsx]: maybe we can rename this to `dimensionFactory`?

**Comment by mvidalgarcia** [comment]
> The only things I noticed so far is a problem I encountered on my Phone (Galaxy S8)
> There is no enough right margin in a few places which makes some elements partially invisible.
> _[...]_

Could you please check now and let me know? It's up on my server

**Pull request merged**